libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -114,6 +114,8 @@ void NavEKF3_core::setWindMagStateLearningMode()
     if (!inhibitMagStates && setMagInhibit) {
         inhibitMagStates = true;
         updateStateIndexLim();
+        // we need to learn variances from scratch
```

Comment:
We shouldn't have to do this here if we are doing the reset in CovariancePrediction.